### PR TITLE
[DBZ-PGYB][yugabyte/yugabyte-db#21394] Test changes for working against YugabyteDB

### PR DESCRIPTION
**Pull request by vaibhav-yb**
This PR includes the changes required for the tests so that they can work against YugabyteDB.

YugabyteDB issue: yugabyte/yugabyte-db#21394

### CODE REVIEW

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java [hunk 1]
```diff
@@ -928,6 +931,10 @@ public void shouldExecuteOnConnectStatements() throws Exception {
         waitForStreamingRunning();
 
         SourceRecords actualRecords = consumeRecordsByTopic(6);
```
Comment (asrinivasanyb) [inline debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java:933]: why is this 6 ?

Comment (asrinivasanyb) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java:933]: Looks like these records are being fetched as part of snapshot. Order is not guaranteed in this case.

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java [hunk 2]
```diff
@@ -1077,9 +1089,11 @@ public void shouldRecoverFromRetriableException() throws Exception {
         // kill all opened connections to the database
         TestHelper.execute("SELECT pg_terminate_backend(pid) FROM pg_stat_activity WHERE backend_type='walsender'");
         TestHelper.execute(INSERT_STMT);
+        TestHelper.waitFor(Duration.ofSeconds(10));
```
Comment (asrinivasanyb) [inline debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java:1092]: Should we reduce the wait duration ?

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java [hunk 3]
```diff
@@ -1562,7 +1601,11 @@ public void shouldRegularlyFlushLsn() throws InterruptedException, SQLException
                 .build();
         start(PostgresConnector.class, config);
         assertConnectorIsRunning();
-        waitForStreamingRunning("postgres", TestHelper.TEST_SERVER);
+//        waitForStreamingRunning("postgres", TestHelper.TEST_SERVER);
+
+        // Wait here for 1 minute.
+        Thread.sleep(60_000);
```
Comment (asrinivasanyb) [inline debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java:1607]: why this 1 minute wait ?

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java [hunk 4]
```diff
@@ -2717,21 +2775,24 @@ public void shouldNotAckLsnOnSource() throws Exception {
         Assert.assertEquals(slotAtTheBeginning.slotLastFlushedLsn(), slotAfterSnapshot.slotLastFlushedLsn());
 
         TestHelper.execute("INSERT INTO s2.a (aa,bb) VALUES (1, 'test');");
-        TestHelper.execute("UPDATE s2.a SET aa=2, bb='hello' WHERE pk=2;");
+        // YB note: since update records are not yet supported, commenting this and reducing the
```
Comment (asrinivasanyb) [inline debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java:2778]: Could you mention the list of tests related to Updates that would need to be enabled once the Update support is made available in the ticket for Update support

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java [hunk 5]
```diff
@@ -3215,6 +3276,7 @@ public void testShouldNotCloseConnectionFetchingMetadataWithNewDataTypes() throw
         System.out.println(recordsForTopic.get(0));
     }
 
+    @Ignore("YB: updating replica identity unsupported")
```
Comment (asrinivasanyb) [inline debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java:3279]: Could you mention the list of tests related to ALTER REPLICA IDENTITY that would need to be enabled once the support for  ALTER REPLICA IDENTITY is made available in the ticket for Update support

**Review by asrinivasanyb — APPROVED** [review]

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java (hunk 3 above)
Comment (vaibhav-yb) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java:1607]: Reduced this to 15 seconds.

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java (hunk 4 above)
Comment (vaibhav-yb) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java:2778]: Done.

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java (hunk 5 above)
Comment (vaibhav-yb) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java:3279]: Done.

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java (hunk 2 above)
Comment (vaibhav-yb) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java:1092]: I have kept it like this to be on the safer side, we could reduce later once we gain more stability.

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java (hunk 1 above)
Comment (vaibhav-yb) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java:938]: Got it, I'll add a separate test similar to this test but without ordering constraint.

**Pull request merged**